lib/rails/generators/slim/component_generator.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require "rails/generators/erb/component_generator"
+
+module Slim
+  module Generators
+    class ComponentGenerator < Erb::Generators::ComponentGenerator
```

Comment:
Why inherit from the ERB version but effectively override everything? Can we just have a true base class and just parameterize the template? In fact do we need classes at all? Can't we just parameterize a single class?

Reply:
Good point, I've simply followed the pattern from `slim-rails` (see https://github.com/slim-template/slim-rails/blob/master/lib/generators/slim/scaffold/scaffold_generator.rb) and honestly did not put much thought into refactoring/cleaning it up. Will take another look!

Reply:
Come to think of it, my thought was to make it easier in the future to move the generators from `actionview-component` to what I thought would be their final destinations (slim-rails and rails).

Reply:
yeah ok. The existing convention is super weird but seems like we should stick with it.

Reply:
I understand the patterns that Rails uses given they have lots of generators. Personally I think slim-rails copying that probably isn't what I'd do. We're not going to upstream this anymore so the value of matching that pattern is reduced. 

I think I'd be tempted to simplify down to a single class that get parameterized but this is generator code some I'm not as bothered about the clean factoring.  If we find we need to keep changing this and making parallel changes across the implementations maybe we refactor then